### PR TITLE
fix(elan/dist): fix bad regex when looking up nightlies with fixed dates

### PR DESCRIPTION
**Pull request by khoek**
Sorry to be a bother---I missed the fact that nightlies with fixed dates get looked up with the wrong name. The problem didn't show up on my machine because I'd inadvertently cached the binary, so it didn't need to be redownloaded. I've just deleted all my toolchains and retested all the cases from scratch now.

Thanks (and sorry) again!

### CODE REVIEW

File: src/elan-dist/src/dist.rs [hunk 1]
```diff
@@ -142,11 +142,12 @@ pub fn update_from_dist<'a>(download: DownloadCfg<'a>,
 }
 
 //A shorthand to return the default origin if a custom one was not specified
-fn fill_origin(origin: Option<&String>) -> String {
-    match origin {
+fn fill_origin(origin: Option<&String>, version: &str) -> String {
```
Comment (Kha) [inline src/elan-dist/src/dist.rs]: Could you adapt the docs to the change and perhaps rename this function?

File: src/elan-dist/src/dist.rs [hunk 2]
```diff
@@ -142,11 +142,12 @@ pub fn update_from_dist<'a>(download: DownloadCfg<'a>,
...
+    let repo = match origin {
         None => DEFAULT_ORIGIN,
         Some (repo) => repo
-    }.to_owned()
+    }.to_owned();
```
Comment (Kha) [inline src/elan-dist/src/dist.rs]: You can remove the `to_owned` now

File: src/elan-dist/src/dist.rs [hunk 3]
```diff
@@ -192,7 +193,7 @@ pub fn update_from_dist_<'a>(download: DownloadCfg<'a>,
         }*/
     }
 
-    match manifestation.update(&fill_origin(toolchain.origin.as_ref()),
+    match manifestation.update(&fill_origin(toolchain.origin.as_ref(), toolchain.channel.as_str()),
```
Comment (Kha) [inline src/elan-dist/src/dist.rs]: `&toolchain.channel` should work as well. Same below.

**Comment by Kha** [comment]
Unfortunately, I did not see your PR before releasing 0.6.0, but I'll just release a 0.6.1 after merging this PR

**Comment by khoek** [comment]
Your comments should now have been applied. I picked "build_origin_name" as the new name---please let me know if you prefer something different!

**Pull request merged**